### PR TITLE
TrackballControls: Introduce custom mouse mappings.

### LINKED ISSUE #7888: Changing TrackballControls STATE
Hi,

I am using **TrackballControls.js** in my project and I am using **bower**. 
I want to change the `PAN` and `ROTATE` values so I can use left click for panning, etc in my project. For that I have to change the `STATE` inside the **TrackballControls.js** which means I lose my changes if I use bower to upgrade the components, etc. 
i.e., I have to change:

```
var STATE = { NONE: - 1, ROTATE: 0, ZOOM: 1, PAN: 2, TOUCH_ROTATE: 3, TOUCH_ZOOM_PAN: 4 };
```

to

```
var STATE = { NONE: -1, ROTATE: 2, ZOOM: 1, PAN: 0, TOUCH_ROTATE: 3, TOUCH_ZOOM_PAN: 4 };
```

Is there anyway to pass the `STATE` variable to `TrackballControl`s instead of changing it inside the code so I can use bower? 

Thanks

### PR DESCRIPTION
**Pull request by Mugen87**
Fixes #7888

The PR introduces `mouseButtons` similar to `OrbitControls`.

### CODE REVIEW

**Comment by Mugen87** [comment]
https://raw.githack.com/Mugen87/three.js/dev36/examples/misc_controls_trackball.html

File: examples/js/controls/TrackballControls.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ THREE.TrackballControls = function ( object, domElement ) {
 
 	this.keys = [ 65 /*A*/, 83 /*S*/, 68 /*D*/ ];
 
+	this.mouseButtons = { LEFT: THREE.MOUSE.LEFT, MIDDLE: THREE.MOUSE.MIDDLE, RIGHT: THREE.MOUSE.RIGHT };
```
Comment (WestLangley) [inline examples/js/controls/TrackballControls.js]: We don't use that pattern anymore. Mouse buttons now specify actions. See #16961.

Comment (Mugen87) [reply examples/js/controls/TrackballControls.js]: Done.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**